Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -19,42 +21,54 @@ public struct SegmentedSwitch<Selection: Hashable, Content: View>: View {
+                        .padding(.vertical, verticalTextPadding)
+                }
+                .backgroundPreferenceValue(IDPreferenceKey.self) { preference in
+                    interactionBackground(preference)
+                }
+                .backgroundPreferenceValue(IDPreferenceKey.self) { preference in
+                    selectionBackground(preference)
+                }
+                .fixedSize(horizontal: false, vertical: true)
             }
         }
-        .accessibilityElement(children: .ignore)
+        .accessibilityElement(children: .contain)
         .accessibility(label: .init(label))
-        .accessibility(value: .init(selection.map(String.init(describing:)) ?? ""))
-        .accessibility(hint: .init(accessibilityHint))
-        .accessibility(addTraits: .isButton)
-        .accessibility(addTraits: selection != nil ? .isSelected : [])
+        .accessibility(hint: .init(message?.description ?? ""))
     }
 
-    @ViewBuilder var separator: some View {
-        (selection == nil ? borderColor : .clear)
-            .frame(width: borderWidth)
-            .frame(maxHeight: .infinity)
-            .padding(.vertical, BorderWidth.active)
+    @ViewBuilder func interactionBackground(_ preferences: [IDPreference]) -> some View {
+        GeometryReader { geometry in
+            ForEach(Array(preferences.enumerated()), id: \.element.id) { index, preference in
+                Color.clear
+                    .contentShape(Rectangle())
+                    .segmentFrame(bounds: preference.bounds, in: geometry)
+                    .onTapGesture {
+                        selection = selection(from: preference)
+                    }
+                    .accessibility(index == 0 ? .segmentedSwitchFirstOption : .segmentedSwitchSecondOption)
+                    .accessibility(value: .init((selection(from: preference)).map(String.init(describing:)) ?? ""))
+                    .accessibility(addTraits: .isButton)
+                    .accessibility(addTraits: isSelected(preference) ? .isSelected : [])
+            }
+        }
     }
 
-    @ViewBuilder func selectionBackground(_ preference: IDPreferenceKey.Value) -> some View {
+    @ViewBuilder func selectionBackground(_ preferences: [IDPreference]) -> some View {
         HStack(spacing: 0) {
-            if let anchor = preference.first(where: { $0.id == selection.map(AnyHashable.init) })?.bounds {
+            if let bounds = preferences.first(where: isSelected)?.bounds {
                 GeometryReader { geometry in
-                    let itemWidth = geometry.size.width / 2
-
                     background
-                        .frame(width: itemWidth)
-                        .offset(x: (geometry[anchor].minX / itemWidth).rounded(.down) * itemWidth)
+                        .segmentFrame(bounds: bounds, in: geometry)
                 }
             } else {
                 background
+                    .overlay(separator)
```

Comment:
To be more precise, the HStack that hosts the content should use `spacing` in the same size of separator, so that the center separator overlay fits exactly between expected two segments.

If a call-site provides anything else than two views, the overlay would be broken. The Tabs-like solution probably can solve this in a follow-up.